testsupport/wait/member.go

```diff
@@ -72,158 +73,197 @@ func (a *MemberAwaitility) WithRetryOptions(options ...RetryOption) *MemberAwait
 }
 
 // UserAccountWaitCriterion a function to check that a user account has the expected condition
-type UserAccountWaitCriterion func(a *MemberAwaitility, ua *toolchainv1alpha1.UserAccount) bool
+type UserAccountWaitCriterion struct {
+	Match func(*toolchainv1alpha1.UserAccount) bool
+	Diff  func(*toolchainv1alpha1.UserAccount) string
+}
+
+func MatchUserAccountWaitCriterion(actual *toolchainv1alpha1.UserAccount, criteria ...UserAccountWaitCriterion) bool {
+	for _, c := range criteria {
+		if !c.Match(actual) {
+			return false
+		}
+	}
+	return true
+}
+
+func SprintUserAccountWaitCriterionDiffs(actual *toolchainv1alpha1.UserAccount, criteria ...UserAccountWaitCriterion) string {
+	buf := &strings.Builder{}
+	buf.WriteString("failed to find UserAccount with matching criteria:\n")
+	for _, c := range criteria {
+		if !c.Match(actual) {
+			buf.WriteString(c.Diff(actual))
+			buf.WriteString("\n")
+		}
+	}
+	return buf.String()
+}
 
 // UntilUserAccountHasSpec returns a `UserAccountWaitCriterion` which checks that the given
 // USerAccount has the expected spec
 func UntilUserAccountHasSpec(expected toolchainv1alpha1.UserAccountSpec) UserAccountWaitCriterion {
-	return func(a *MemberAwaitility, ua *toolchainv1alpha1.UserAccount) bool {
-		a.T.Logf("waiting for useraccount specs. Actual: '%+v'; Expected: '%+v'", ua.Spec, expected)
-		userAccount := ua.DeepCopy()
-		userAccount.Spec.NSTemplateSet = toolchainv1alpha1.NSTemplateSetSpec{}
-		expectedSpec := expected.DeepCopy()
-		expectedSpec.NSTemplateSet = toolchainv1alpha1.NSTemplateSetSpec{}
-		return reflect.DeepEqual(userAccount.Spec, *expectedSpec)
+	return UserAccountWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.UserAccount) bool {
+			userAccount := actual.DeepCopy()
+			userAccount.Spec.NSTemplateSet = toolchainv1alpha1.NSTemplateSetSpec{}
+			expectedSpec := expected.DeepCopy()
+			expectedSpec.NSTemplateSet = toolchainv1alpha1.NSTemplateSetSpec{}
+			return reflect.DeepEqual(userAccount.Spec, *expectedSpec)
+		},
```

Comment:
adding to rajiv's finding, missing a diff here too?

Reply:
done in 9085b05